src/Algebra/Graph/Bipartite/AdjacencyMap.hs

```diff
@@ -0,0 +1,165 @@
+-- different parts are equal, these vertices are considered to be different.
+-- See the examples for more details.
+data AdjacencyMap a b = BAM {
+    -- | The /adjacency map/ of the left part of the graph: each vertex is
+    -- associated with a set of its neighbours. Complexity: /O(1)/ time and
+    -- memory
+    --
+    -- @
+    -- leftToRight ('fromGraph' 'Algebra.Graph.empty')                                          == Map.'Map.empty'
+    -- leftToRight ('fromGraph' ('Algebra.Graph.vertex' (Left 1)))                              == Map.'Map.singleton' 1 Set.'Set.empty'
+    -- leftToRight ('fromGraph' ('Algebra.Graph.vertex' (Right 1)))                             == Map.'Map.empty'
+    -- leftToRight ('fromGraph' ('Algebra.Graph.edge' (Left 1) (Right 2)))                      == Map.'Map.singleton' 1 (Set.'Set.singleton' 2)
+    -- leftToRight ('fromGraph' ('Algebra.Graph.edge' (Left 1) (Right 1)))                      == Map.'Map.singleton' 1 (Set.'Set.singleton' 1)
+    -- leftToRight ('fromGraph' ('Algebra.Graph.edges' [(Left 1, Right 1), (Left 1, Right 2)])) == Map.'Map.singleton' 1 (Set.'Set.fromAscList' [1, 2])
+    -- @
+    leftToRight :: Map.Map a (Set.Set b),
+
+    -- | The inverse map for 'leftToRight'. Complexity: /O(1)/ time and memory.
+    --
+    -- @
+    -- rightToLeft ('fromGraph' 'Algebra.Graph.empty')                                          == Map.'Map.empty'
+    -- rightToLeft ('fromGraph' ('Algebra.Graph.vertex' (Left 1)))                              == Map.'Map.empty'
+    -- rightToLeft ('fromGraph' ('Algebra.Graph.vertex' (Right 1)))                             == Map.'Map.singleton' 2 Set.'Set.empty'
+    -- rightToLeft ('fromGraph' ('Algebra.Graph.edge' (Left 1) (Right 2)))                      == Map.'Map.singleton' 2 (Set.'Set.singleton' 1)
+    -- rightToLeft ('fromGraph' ('Algebra.Graph.edge' (Left 1) (Right 1)))                      == Map.'Map.singleton' 1 (Set.'Set.singleton' 1)
+    -- rightToLeft ('fromGraph' ('Algebra.Graph.edges' [(Left 1, Right 1), (Left 1, Right 2)])) == Map.'Map.fromAscList' [(1, Set.'Set.singleton' 1), (2, Set.'Set.singleton' 1)]
+    -- @
+    rightToLeft :: Map.Map b (Set.Set a)
+} deriving (Eq, Show)
+
+-- | Check that the internal graph representation is consistent, i.e. that all
+-- edges that are present in the 'leftToRight' map are present in the
+-- 'rightToLeft' map.
+-- Complexity: /O(m log m)/
+--
+-- @
+-- consistent ('fromGraph' 'Algebra.Graph.empty')      == True
+-- consistent ('fromGraph' ('Algebra.Graph.vertex' x)) == True
+-- consistent ('fromGraph' ('Algebra.Graph.edge' x y)) == True
+-- consistent ('fromGraph' ('Algebra.Graph.edges' xs)) == True
+-- consistent ('fromGraph' g)          == True
+-- consistent ('fromAdjacencyMap' am)  == True
+-- @
+consistent :: (Ord a, Ord b) => AdjacencyMap a b -> Bool
+consistent (BAM lr rl) = internalEdgeList lr == sort (map swap $ internalEdgeList rl)
+
+-- | The list of edges of a bipartite adjacency map
```

Comment:
The convention in the library is to not give Haddock comments to functions that are not exported -- it makes it easier to immediately see whether a function is exported or not, so please change `-- |` to just `--`.

Reply:
Same for `addReverseEdges`.